src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -674,6 +674,31 @@ internal RankingTrainers(RankingCatalog catalog)
             var eval = new RankingEvaluator(Environment, options ?? new RankingEvaluatorOptions() { });
             return eval.Evaluate(data, labelColumnName, rowGroupColumnName, scoreColumnName);
         }
+
+        /// <summary>
+        /// Run cross-validation over <paramref name="numberOfFolds"/> folds of <paramref name="data"/>, by fitting <paramref name="estimator"/>,
+        /// and respecting <paramref name="rowGroupColumnName"/>if provided.
+        /// Then evaluate each sub-model against <paramref name="labelColumnName"/> and return metrics.
+        /// </summary>
+        /// <param name="data">The data to run cross-validation on.</param>
+        /// <param name="estimator">The estimator to fit.</param>
+        /// <param name="numberOfFolds">Number of cross-validation folds.</param>
+        /// <param name="labelColumnName">The label column (for evaluation).</param>
+        /// <param name="rowGroupColumnName">The name of the groupId column in <paramref name="data"/>, which is used to group rows.
+        /// While for other crossvalidation methods this column is called samplingKeyColumnName, ranking requires
+        /// this column to be <paramref name="rowGroupColumnName"/>.
```

Comment:
Nit: I think saying the following is a little bit misleading: "... for other crossvalidation methods this column is called samplingKeyColumnName ..." and would prefer if the whole comment was rewritten to something like "This column will automatically be used as SamplingKeyColumn when splitting the data for Cross Validation, as this is required by the ranking algorithms".

As discussed offline, we might be adding another overload for CV that will let the user set both `samplingKeyColumnName` and `rowGroupColumnNameName` separately; the fact that we are only providing the `rowGroupColumnName` here doesn't mean "it's called samplingKeyColumnName" in the CV for the other tasks.

Reply:
Note: This comment has remained unaddressed

---
In reply to: [451955399](https://github.com/dotnet/machinelearning/pull/5263#discussion_r451955399) [](ancestors = 451955399)